### PR TITLE
doc related changes

### PR DESCRIPTION
**Pull request by japaric**
see individual commit messages for details

### CODE REVIEW

**Comment by japaric** [comment]
one sec. It seems that docs.rs has some problems with links into dependencies. Will re-open this after double checking that.

File: src/lib.rs [hunk 1]
```diff
@@ -16,19 +16,18 @@
...
-//! - [`#[exception]`] to override an exception handler. If not overridden all exception handlers
+//! - `#[entry]` to declare the entry point of the program
+//! - `#[exception]` to override an exception handler. If not overridden all exception handlers
 //!   default to an infinite loop.
-//! - [`#[pre_init]`] to run code *before* `static` variables are initialized
-//!
-//! [`#[entry]`]: ../cortex_m_rt_macros/fn.entry.html
-//! [`#[exception]`]: ../cortex_m_rt_macros/fn.exception.html
-//! [`#[pre_init]`]: ../cortex_m_rt_macros/fn.pre_init.html
+//!   default to an infinite loop.
```
Comment (adamgreig) [inline src/lib.rs]: You have this line twice for some reason.

Comment (japaric) [reply src/lib.rs]: removed

**Pull request merged**